src/js/preshuffle.s

```diff
@@ -1423,6 +1424,112 @@ DynaShoot2:
 
 .segment "fe", "ff"
 
+; .ifdef _CROWD_CONTROL_
+;;; Hook into the main loop right after the other hooks
+;;; so it shouldn't affect anything else (this is only after input is read)
+.org $cb68
+  jsr HandleCrowdControl
+
+.reloc
+HandleCrowdControl:
+  lda CrowdControlFlag
+  ; check if theres even anything in the queue
+  beq @continue
+    lda CrowdControlQueue
+    ldx #$08 ; we pre decrement so start at 8
+    ; check each bit one at a time for sfx
+@loop:
+      dex
+      bmi @finalize
+      asl
+      bcs @handle_op
+      bcc @loop
+@handle_op:
+  pha
+    ; x has the offset for the crowd control main function
+    txa
+    pha
+      asl
+      tay
+      ; $10,11,12 is used for jump points, it seems to be always overwritten
```

Comment:
There are better ways to do a dynamic JSR.  For instance, this one saves 5 bytes and doesn't use any RAM:

```
; ...
  jsr CCMainOp
; ...

.reloc
CCMainOp:
  lda CCMainOpTable,y ; note: I may have this backwards
  pha
  lda CCMainOpTable+1,y
  pha
  rts
```

One of the main benefits of avoiding RAM is that it's safer w.r.t. interrupts (i.e. you don't need to worry about whether you're in the interrupt handler or not, since the handlers aren't actually allowed to touch $10..12, for obvious reasons).